compiler/noirc_frontend/src/parser/parser/expression.rs

```diff
@@ -502,20 +502,39 @@ impl Parser<'_> {
     }
 
     fn parse_constructor_field(&mut self) -> Option<(Ident, Expression)> {
-        let ident = self.eat_ident()?;
-
-        Some(if self.eat_colon() {
-            let expression = self.parse_expression_or_error();
-            (ident, expression)
-        } else if self.at(Token::DoubleColon) || self.at(Token::Assign) {
-            // If we find '='  or '::' instead of ':', assume the user meant ':`, error and continue
-            self.expected_token(Token::Colon);
-            self.bump();
-            let expression = self.parse_expression_or_error();
-            (ident, expression)
-        } else {
-            (ident.clone(), ident.into())
-        })
+        // Loop to do some error recovery
+        loop {
+            // Make sure not to loop forever
+            if self.at_eof() || self.at(Token::RightBrace) {
+                return None;
+            }
+
+            // If we can't find an identifier, error but continue looking for an identifier
+            if !matches!(self.token.token(), Token::Ident(..)) {
+                self.expected_identifier();
+                self.bump();
+                // Don't error again if a comma comes next
+                if self.at(Token::Comma) {
+                    self.bump();
+                }
+                continue;
+            }
+
+            let ident = self.eat_ident()?;
+
+            return Some(if self.eat_colon() {
+                let expression = self.parse_expression_or_error();
+                (ident, expression)
+            } else if self.at(Token::DoubleColon) || self.at(Token::Assign) {
+                // If we find '='  or '::' instead of ':', assume the user meant ':`, error and continue
+                self.expected_token(Token::Colon);
+                self.bump();
+                let expression = self.parse_expression_or_error();
+                (ident, expression)
```

Comment:
For completeness, can you add some error cases for this?

Reply:
You mean, when we find '=' or '::' instead of ':'? If so, that was done in a previous PR (some months ago) and it's already tested. I think it shows up in the diff because everything was put inside a loop.